app.js

```diff
@@ -1,458 +1,573 @@
+               const bookType = book.repository_name ? 'github' : (book.collection_id ? 'legacy' : null)
+               if (!bookType) continue
+               const legacyBook = bookType === 'legacy'
+               let totalMatches = 0
+               isSkipping = false
+               
+               const t1 = document.createElement('li')
+               const detailsEl = document.createElement('details')
+               const summaryEl = document.createElement('summary')
+               const bookResultsEl = document.createElement('ul')
+               const bookLogEl = document.createElement('span')
+               const bookTitleEl = document.createElement('span')
+               bookLogEl.classList.add('book-status')
+               t1.append(detailsEl)
+               detailsEl.append(summaryEl)
+               summaryEl.append(bookTitleEl)
+               summaryEl.append(bookLogEl)
+               
+               detailsEl.append(bookResultsEl)
+               resultsEl.prepend(t1)
+               
+               let bookUrl = legacyBook ? `${legacyServerRootUrl}/${book.books[0].uuid}` : null
+               let bookJson = null
+               let collectionXMLLinks
+               if (!legacyBook) {
+                    //Download the book.xml from Github
+                    bookXML = await fetchWithBackoff(`${openstaxGithubURL}/${book.repository_name}/META-INF/books.xml`, false)
+                    sandboxEl.innerHTML = bookXML.replace(/ src=/g, ' data-src=')
+                    //List of collection.xml files
+                    collectionXMLLinks = Array.from(sandboxEl.querySelectorAll("book[href]"), e => e.attributes.getNamedItem('href').value)
+                    const pipelines = JSON.parse(window.localStorage.getItem('pipelines'))
+                    const commit_shas = []
+                    book.versions.forEach(v => commit_shas.push(v.commit_sha.substring(0, 7)))
+                    for (let pipeline of pipelines) {
+                         for (let commitSha of commit_shas) {
+                              bookUrl = `${s3RootUrl}/${pipeline}/contents/${bookUuid}@${commitSha}.json`
+                              try {
+                                   //Find the right book.json file and exit the loop
+                                   bookJson = await fetchWithBackoff(bookUrl, true)
+                                   if (bookJson && bookJson.title) break
+                              } catch (error) {
+                                   console.error(error)
+                                   continue;
+                              }
+                         }
+                         if (bookJson && bookJson.title) break
+                    }
+               } else bookJson = await fetchWithBackoff(bookUrl, true)
+               
+               
+               analyzeFn('BOOK:START', bookJson, bookUuid)
+               
+               
+               bookTitleEl.textContent = bookJson.title
```

Comment:
If a valid codeversion cannot be found `bookJson` is null.
```suggestion
               if (bookJson === null) {
                   alert(`Could not find a baked version of ${bookUuid}@${commitSha}. Tried using these codeversions: ${JSON.stringify(pipelines)}`)
                   return
               }
               bookTitleEl.textContent = bookJson.title
```